test/test_utils.dart

```diff
@@ -0,0 +1 @@
+int generateTestingPortFromName(String name) => 50000 + (name.hashCode % 10000);
```

Comment:
do we actually need this function? why not just hardcode a port per test?

Reply:
we tossed that around as an idea, I prefer an approach like this because we will have a lot of tests and manually tracking which ports have been used could be prone to errors.

Reply:
cool, as for the name - nothing else comes to mind - maybe it can be shorted to just `generateTestingPort` for brevity's sake?

```suggestion
int generateTestingPort(String name) => 50000 + (name.hashCode % 10000);
```